Integration test 'integration-basic' fails when run against an existing Spark bundle deployment

When the `integration-basic` test is run against an existing Spark bundle, the test fails. It is because in the bundle, an extra service account is created due to Kyuubi <> Integration Hub relation, while the test asserts there's only one service account created via fixture.

**Steps to reproduce:**
1. Deploy Spark K8s bundle.
2. Run the `integration-basic` test as:
```
tox -e integration-basic
```

**Observation**
The test fails with an Assertion error
```
_______________________________________________________________________________________ test_pod_admin _______________________________________________________________________________________
Traceback (most recent call last):
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/runner.py", line 241, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 182, in _multicall
    return outcome.get_result()
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_result.py", line 100, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 167, in _multicall
    teardown.throw(outcome._exception)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/threadexception.py", line 87, in pytest_runtest_call
    yield from thread_exception_runtest_hook()
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/threadexception.py", line 63, in thread_exception_runtest_hook
    yield
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 167, in _multicall
    teardown.throw(outcome._exception)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/unraisableexception.py", line 90, in pytest_runtest_call
    yield from unraisable_exception_runtest_hook()
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/unraisableexception.py", line 65, in unraisable_exception_runtest_hook
    yield
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 167, in _multicall
    teardown.throw(outcome._exception)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/logging.py", line 850, in pytest_runtest_call
    yield from self._runtest_for(item, "call")
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/logging.py", line 833, in _runtest_for
    yield
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 167, in _multicall
    teardown.throw(outcome._exception)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/capture.py", line 878, in pytest_runtest_call
    return (yield)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 167, in _multicall
    teardown.throw(outcome._exception)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/skipping.py", line 257, in pytest_runtest_call
    return (yield)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/runner.py", line 183, in pytest_runtest_call
    raise e
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/runner.py", line 173, in pytest_runtest_call
    item.runtest()
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 182, in _multicall
    return outcome.get_result()
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_result.py", line 100, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/home/ubuntu/spark-k8s-bundle/python/.tox/integration-basic/lib/python3.10/site-packages/_pytest/python.py", line 162, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/ubuntu/spark-k8s-bundle/python/tests/integration/test_basic.py", line 38, in test_pod_admin
    assert output.strip("\n") == service_account.id
AssertionError: assert 'spark-ns:spark-test-e06116c1-ccfa-4e1c-9c18-c635e8e837ad\nspark:kyuubi-engine' == 'spark-ns:spark-test-e06116c1-ccfa-4e1c-9c18-c635e8e837ad'
  
  - spark-ns:spark-test-e06116c1-ccfa-4e1c-9c18-c635e8e837ad
  + spark-ns:spark-test-e06116c1-ccfa-4e1c-9c18-c635e8e837ad
  ?                                                         +
  + spark:kyuubi-engine
```

**Expected result**
The test passes without failures.

Additional context
[This particular assertion](https://github.com/canonical/spark-k8s-bundle/blob/main/python/tests/integration/test_basic.py#L38) is causing the test to fail, because it fails to account for a service account that's automatically created due to Spark <> Integration Hub relation.

--- a/python/tests/integration/test_basic.py
+++ b/python/tests/integration/test_basic.py
@@ -30,12 +30,11 @@ def namespace_name():
 
 
 def test_pod_admin(admin_pod, service_account):
-
     output: str = admin_pod.exec(
         ["spark-client.service-account-registry", "list"]
     ).decode("utf-8")
 
-    assert output.strip("\n") == service_account.id
+    assert service_account.id in output.splitlines()
 
 
 def test_spark_submit(
@@ -46,7 +45,6 @@ def test_spark_submit(
     small_profile_properties,
     image_properties,
 ):
-
     extra_confs = small_profile_properties + image_properties
 
     registry.set_configurations(service_account.id, extra_confs)

